runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7183,6 +7183,52 @@ done:;
 		return rc;
 	}
 
+	/*
+	 * Determine the equality of two objects
+	 * @param[in] lhs the lhs object of acmp bytecodes
+	 * @param[in] rhs the rhs object of acmp bytecodes
+	 * return true if they are equal and false otherwise
+	 */
+	VMINLINE bool
+	acmpSub(j9object_t lhs, j9object_t rhs)
+	{
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		bool acmpResult = false;
+		if (rhs == lhs) {
+			acmpResult = true;
+		} else if ((NULL == lhs) || (NULL == rhs)) {
+			acmpResult = false;
+		} else {
+			J9Class * lhsClass = J9OBJECT_CLAZZ(_currentThread, lhs);
+			J9Class * rhsClass = J9OBJECT_CLAZZ(_currentThread, rhs);
+			if(J9_IS_J9CLASS_VALUETYPE(lhsClass)
```

Comment:
needs to be a space between `if` and `(`

Reply:
since you have already gotten the class, you can add another contition that says `lhsClass == rhsClass` before checking if they are substitutable

Reply:
please review. I added a condition that check if they are substitutable when `lhsClass == rhsClass` and return false directly when `lhsClass != rhsClass`, so function isSubstitutable() will be called with precondition that `lhsClass == rhsClass`.